api/src/controllers/planDeTransport/import.ts

```diff
@@ -1,40 +1,91 @@
+import fs from "fs";
+import { parse as parseDate } from "date-fns";
+import fileUpload, { UploadedFile } from "express-fileupload";
+import {
+  CohesionCenterModel,
+  PointDeRassemblementModel,
+  ImportPlanTransportModel,
+  LigneBusModel,
+  SessionPhase1Model,
+  LigneToPointModel,
+  PlanTransportModel,
+  ClasseModel,
+  CohortModel,
+  type ReferentDocument,
+  type PlanTransportModesType,
+} from "../../models";
+
 const scanFile = require("../../utils/virusScanner");
-const { getMimeFromFile } = require("../../utils/file");
-const { validateId } = require("../../utils/validator");
+import { getMimeFromFile } from "../../utils/file";
+import { validateId } from "../../utils/validator";
+import { validatePdtFile, computeImportSummary } from "../../planDeTransport/planDeTransport/import/pdtImportService";
+import { formatTime } from "../../planDeTransport/planDeTransport/import/pdtImportUtils";
+import { startSession, withTransaction, endSession } from "../../mongo";
+
+interface ImportPlanTransportLine {
+  [key: string]: string | string[] | undefined;
+  "NUMERO DE LIGNE": string;
+  "DATE DE TRANSPORT ALLER": string;
+  "DATE DE TRANSPORT RETOUR": string;
+  "ID CENTRE": string;
+  "HEURE D'ARRIVEE AU CENTRE": string;
+  "HEURE DE DÉPART DU CENTRE": string;
+  "TOTAL ACCOMPAGNATEURS": string;
+  "CAPACITÉ VOLONTAIRE TOTALE": string;
+  "CAPACITE TOTALE LIGNE": string;
+  "PAUSE DÉJEUNER ALLER"?: string;
+  "PAUSE DÉJEUNER RETOUR"?: string;
+  "TEMPS DE ROUTE": string;
+  "ID CLASSE"?: string;
+  "LIGNES FUSIONNÉES"?: string;
+}
+
+interface StepPoint {
+  type: string;
+  address: string;
+  departureHour: string;
+  returnHour: string;
+  transportType: string;
+}
+
+interface LineToPoint {
+  lineId: string;
+  meetingPointId: string;
+  transportType: string;
+  busArrivalHour: string;
+  departureHour: string;
+  meetingHour: string;
+  returnHour: string;
+  stepPoints: StepPoint[];
+}
 
-const { validatePdtFile, computeImportSummary } = require("../../planDeTransport/planDeTransport/import/pdtImportService");
-const { formatTime } = require("../../planDeTransport/planDeTransport/import/pdtImportUtils");
-const { startSession, withTransaction, endSession } = require("../../mongo");
+interface CustomRequestWithFiles extends Request {
+  user: Partial<ReferentDocument>;
+  files?: {
+    [fieldname: string]: UploadedFile | UploadedFile[];
+  };
+}
+
+const router: Router = express.Router();
 
 // Vérifie un plan de transport importé et l'enregistre dans la collection importplandetransport.
 router.post(
   "/:cohortName",
   passport.authenticate("referent", { session: false, failWithError: true }),
   fileUpload({ limits: { fileSize: 5 * 1024 * 1024 }, useTempFiles: true, tempFileDir: "/tmp/" }),
-  async (req, res) => {
+  async (req: CustomRequestWithFiles, res: Response) => {
```

Comment:
```suggestion
  async (req: UserRequest, res: Response) => {
```
`/api/src/controllers/request.ts`